src/Microsoft.AspNetCore.Mvc.DataAnnotations/Internal/DataAnnotationsMetadataProvider.cs

```diff
@@ -210,7 +220,7 @@ public void CreateDisplayMetadata(DisplayMetadataProviderContext context)
             // TemplateHint
             if (uiHintAttribute != null)
             {
-                displayMetadata.TemplateHint = uiHintAttribute.UIHint;
+                displayMetadata.TemplateHint = LocalizeAttributeProperty(() => { return uiHintAttribute.UIHint; }, localizer)();
```

Comment:
Remove the cruft from the argument expression. Use either `uiHintAttribute.UIHint` (which the compiler might forgive) or `() => uiHintAttribute.UIHint` (which definitely works).